Modules/unicodedata.c

```diff
@@ -1253,6 +1274,112 @@ unicodedata_UCD_lookup_impl(PyObject *self, const char *name,
     return PyUnicode_FromOrdinal(code);
 }
 
+typedef struct {
+    PyObject_HEAD
+    PyObject* str;
+    int pos;
+} GraphemeClusterIterator;
+
+void GCI_Del(PyObject* x)
+{
+    GraphemeClusterIterator *i = (GraphemeClusterIterator *)x;
+    Py_DECREF(i->str);
+    PyObject_Del(x);
+}
+
+static PyObject* GCI_iter(PyObject *self)
+{
+    Py_INCREF(self);
+    return self;
+}
+
+#include "grapheme_cluster_break_automaton.h"
+
+PyObject* GCI_iternext(PyObject *self)
+{
+    GraphemeClusterIterator *p = (GraphemeClusterIterator *)self;
+    int kind = PyUnicode_KIND(p->str);
+    void *pstr = PyUnicode_DATA(p->str);
+    if (PyUnicode_READ(kind, pstr, p->pos)) {
```

Comment:
Why not? NUL characters don't have special meaning in Python strings.

Silent truncating the string on the first NUL character would look like a bug.